sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -488,7 +491,9 @@ export class MessageSession extends LinkEntity {
         this._context.namespace.connectionId
       );
 
-      if (isSessionExpired === undefined || !isSessionExpired) {
+      if (this.isClosedDueToExpiry) {
+        this.isClosedDueToExpiry = false;
```

Comment:
Why are we setting this to `false` here?

Reply:
Because it's true only when there's a `SessionLockLostError`. Once that condition is through, the flag needs to be reset for subsequent close() invocations to work as expected.
I recall specifically needing to add this based on test results.

Reply:
In that case, I would prefer to not have `isClosedDueToExpiry` as property on the class.
I actually liked the the previous version, where you had this as an argument for the close() method. 
Back then, the only request was for a name change.

At a quick glance at the code in this section now, it gets confusing as to how the state of "closed due to session expiry" ever change? i.e this suggests (at a quick glance, you can imagine especially in the future when new developers try to work on this) that when a session is expired, it has the possibility of getting to a non-expired state.

In order to simplify the tracking of `isClosedDueToExpiry`, can we please have the previous model of passing it as an optional parameter?

The only methods that care about this value are `_onAmqpClose` and `close()`, we can go